### PR TITLE
Add UnitTest for Volume

### PR DESCRIPTION
**Pull request by FabianKlimpel**
This PR adds a unit test for the `Volume` class.

### CODE REVIEW

File: Tests/UnitTests/Core/Geometry/VolumeTests.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  Volume volumeShift(volume, &shift);
+  BOOST_TEST(volumeShift.center() ==
+             (volume.transform() * shift).translation());
+  BOOST_TEST(volumeShift.transform().rotation() ==
+             volume.transform().rotation());
+
+  // Test a bounding box from the volume
+  // Vector3D envelope{7_mm, 8_mm, 9_mm};
+  // BOOST_TEST(volume.boundingBox(envelope) == bounds.boundingBox(&transform,
+  // envelope, &volume));
```
Comment (Corentin-Allaire) [inline Tests/UnitTests/Core/Geometry/VolumeTests.cpp:71]: I am not completely sure why you commented out the bounding box test but you should either add it back or completely remove it if it does not work.

Comment (paulgessinger) [reply Tests/UnitTests/Core/Geometry/VolumeTests.cpp:71]: @FabianKlimpel can you comment here?

Comment (paulgessinger) [reply Tests/UnitTests/Core/Geometry/VolumeTests.cpp:71]: I don't think I implemented the equality operator to work this way here. If you'd want to test this you'd have to manually assert that the two defining vertices are float equivalent.

Comment (FabianKlimpel) [reply Tests/UnitTests/Core/Geometry/VolumeTests.cpp:71]: I will remove it for now. It is not implemented but will I added the operator for the #272.

Comment (FabianKlimpel) [reply Tests/UnitTests/Core/Geometry/VolumeTests.cpp]: Gone

**Pull request merged**